app/charts/bar/bars-stacked-state.tsx

```diff
@@ -174,53 +174,53 @@ const useBarsStackedState = (
     getSegment,
   ]);
 
-  const sumsByX = useMemo(() => {
+  const sumsByY = useMemo(() => {
     return Object.fromEntries(
       rollup(
         chartData,
-        (v) => sum(v, (x) => getY(x)),
-        (x) => getX(x)
+        (v) => sum(v, (x) => getX(x)),
+        (x) => getY(x)
       )
     );
   }, [chartData, getX, getY]);
 
   const normalize = calculationType === "percent";
-  const chartDataGroupedByX = useMemo(() => {
+  const chartDataGroupedByY = useMemo(() => {
     if (normalize) {
       return group(
-        normalizeData(chartData, {
-          yKey: yMeasure.id,
-          getY,
-          getTotalGroupValue: (d) => sumsByX[getX(d)],
+        normalizeDataInverted(chartData, {
+          xKey: xMeasure.id,
+          getX,
+          getTotalGroupValue: (d) => sumsByY[getY(d)],
         }),
-        getX
+        getY
       );
     }
 
-    return group(chartData, getX);
-  }, [chartData, getX, sumsByX, getY, yMeasure.id, normalize]);
+    return group(chartData, getY);
+  }, [chartData, getX, sumsByY, getY, xMeasure.id, normalize]);
 
   const chartWideData = useMemo(() => {
-    return getWideData({
-      dataGroupedByX: chartDataGroupedByX,
-      xKey,
-      getY,
+    return getWideDataInverted({
```

Comment:
Nit / question: Here I also started to wonder if we couldn't try to generalize the function a bit with same pattern as above proposal (`valueKey`, `getValue`, `groupedData`)? I am not sure if the logic between `getWideData` and `getWideDataInverted` is actually different, or we just use different variables?